### PR TITLE
Fix missing case for memory

### PR DESCRIPTION
**Pull request by dalehamel**
@dwradcliffe I've overhauled this to remove a lot of magic, and use proper dmi type keys.

### CODE REVIEW

File: lib/genesis_collector/dmidecode.rb [hunk 1]
```diff
@@ -14,10 +59,11 @@ def parse_dmidecode(data)
         case line
         when /^End Of Table/, /^\s+$/, /^\# dmidecode/, /^SMBIOS/, /structures occupying/, /^Table at/
           next
-        when /^Handle\s+(.*?),\s+/
+        when /^Handle\s+(.*?), DMI type (.*?),\s+/
           handle = $1.to_i(16)
-        when /(.*)\s+Information\n$/, /(.*)\s+Device\n$/, /(.*)\s+Device Mapped Address\n$/, /(.*)\s+Array Mapped Address\n$/
-          title = standardize_dmi_key($1)
+          type_id = $2.to_i
+          next if type_id == 127
```
Comment (ibawt) [inline lib/genesis_collector/dmidecode.rb]: 127 in this context is what? other than max signed 8 bit integer

Comment (dalehamel) [inline lib/genesis_collector/dmidecode.rb]: end of table (fixed)

**Comment by dwradcliffe** [comment]
LGTM

File: lib/genesis_collector/dmidecode.rb [hunk 2]
```diff
@@ -14,10 +60,11 @@ def parse_dmidecode(data)
         case line
         when /^End Of Table/, /^\s+$/, /^\# dmidecode/, /^SMBIOS/, /structures occupying/, /^Table at/
           next
-        when /^Handle\s+(.*?),\s+/
+        when /^Handle\s+(.*?), DMI type (.*?),\s+/
           handle = $1.to_i(16)
-        when /(.*)\s+Information\n$/, /(.*)\s+Device\n$/, /(.*)\s+Device Mapped Address\n$/, /(.*)\s+Array Mapped Address\n$/
-          title = standardize_dmi_key($1)
+          type_id = $2.to_i
```
Comment (ibawt) [inline lib/genesis_collector/dmidecode.rb]: just hating on the globals here not that it matters in rubby.

**Pull request merged**